app/jobs/reports/idv_legacy_conversion_supplement_report.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Reports
+  class IdvLegacyConversionSupplementReport < BaseReport
+    REPORT_NAME = 'idv-legacy-conversion-supplement-report'
+
+    def perform(_date)
+      csv = build_csv
+      save_report(REPORT_NAME, csv, extension: 'csv')
+    end
+
+    # @return [String] CSV report
+    def build_csv
+      results = Agreements::IaaOrder.joins(integrations: :service_provider).joins(:iaa_gtc).
+        joins(
+          <<-SQL,
+            INNER JOIN (
+              SELECT DISTINCT ON (user_id) *
+              FROM sp_upgraded_biometric_profiles
+            ) sp ON sp.issuer = integrations.issuer
+          SQL
+        ).select(
+          <<-SQL,
+            iaa_orders.*,
+            iaa_gtcs.gtc_number AS gtc_number,
+            sp.issuer AS issuer,
+            service_providers.friendly_name AS friendly_name,
+            DATE_TRUNC('month', sp.upgraded_at) AS year_month,
+            count(distinct sp.user_id) AS user_count
+          SQL
+        ).where(
+          'sp.upgraded_at BETWEEN iaa_orders.start_date AND iaa_orders.end_date',
+        ).group('iaa_orders.id, sp.issuer, year_month, iaa_gtcs.gtc_number,
+          service_providers.friendly_name').
+        order('iaa_orders.id, year_month')
```

Comment:
I feel like this might be easier as one big `<<~SQL` query instead of mixing in ActiveRecord like this?

Reply:
Yeah, this was just the outgrowth of starting with AR. I wasn't sure if I was just gonna SQL it or throw in #{subquery} here and there.